src/pages/instances/forms/UploadExternalFormatFileForm.tsx

```diff
@@ -0,0 +1,333 @@
+    });
+
+    if (current === total) {
+      close();
+      toastNotify.info(
+        <>
+          Upload completed. Now creating instance{" "}
+          <strong>{instanceName}</strong>.
+        </>,
+      );
+      navigate(`/ui/project/${project?.name}/instances`);
+    }
+  };
+
+  const handleUploadError = (error: Error) => {
+    toastNotify.failure("Image upload failed.", error);
+  };
+
+  const handleSuccess = (instanceName: string) => {
+    const message = (
+      <>
+        Created instance{" "}
+        <InstanceLink
+          instance={{ project: project?.name || "", name: instanceName }}
+        />
+        .
+      </>
+    );
+
+    const actions = [
+      {
+        label: "Configure",
+        onClick: () =>
+          navigate(
+            `/ui/project/${project?.name}/instance/${instanceName}/configuration`,
+          ),
+      },
+    ];
+
+    toastNotify.success(message, actions);
+  };
+
+  const handleFailure = (msg: string) => {
+    toastNotify.failure("Instance creation failed.", new Error(msg));
+  };
+
+  const invalidateCache = () => {
+    void queryClient.invalidateQueries({
+      predicate: (query) => {
+        return query.queryKey[0] === queryKeys.instances;
+      },
+    });
+  };
+
+  const handleConfirmUpload = (values: UploadExternalFormatFileFormValues) => {
+    if (!values.imageFile) {
+      return;
+    }
+
+    // start create instance operation
+    createInstance(uploadExternalFormatFilePayload(values), project?.name || "")
+      .then((operation) => {
+        const operationId = operation.metadata.id;
+        const operationSecret = operation.metadata?.metadata?.fs;
+        const instanceName = getInstanceName(operation.metadata);
+
+        // establish websocket connection based on the instance creation operation
+        let wsUrl = `wss://${location.host}/1.0/operations/${operationId}/websocket`;
+        if (operationSecret) {
+          wsUrl += `?secret=${operationSecret}`;
+        }
+
+        const ws = new WebSocket(wsUrl);
+
+        ws.onopen = () => {
+          sendFileByWebSocket(
+            ws,
+            formik.values.imageFile,
+            FILE_CHUNK_SIZE,
+            handleUploadProgress.bind(null, instanceName),
+            handleUploadError,
+          );
+        };
+
+        // set up event queue for the operation
+        eventQueue.set(
+          operationId,
+          () => handleSuccess(getInstanceName(operation.metadata)),
+          handleFailure,
+          invalidateCache,
+        );
+      })
+      .catch((e) => {
+        close();
+        toastNotify.failure("Instance creation failed.", e);
+      });
+  };
+
+  const formik = useFormik<UploadExternalFormatFileFormValues>({
+    initialValues: {
+      name: "",
+      pool: "",
+      imageFile: null,
+      formatConversion: true,
+      virtioConversion: false,
+      architecture: supportedVMArchOptions(
+        settings?.environment?.architectures || [],
+      )[0].value,
```

Comment:
Can we use the `archOptions` variable, defined below? Probably can move that up a bit in the file and use it here.